SU2_CFD/src/solver_structure.cpp

```diff
@@ -4825,6 +4825,365 @@ void CSolver::Read_InletFile_ASCII(CGeometry *geometry, CConfig *config, string
+
+                          break;
+                        
+                        case(ALPHA_PHI):
+                          Alpha=Parameter1;
+                          Phi=Parameter2;
+                          unit_m = sqrt(1/(1+pow(tan(Alpha),2)));
+
+                          unit_Theta = tan(Alpha)*unit_m;
+                          unit_r=unit_m*sin(Phi);
+
+                          break;
+                        
+                        case(BETA_PHI):
+                              cout<<"In Beta Gamma case\n";
+                              
+                              //get Rotation rate from config file after merging with develop
+                              //convert WTheta to Vtheta
+
+                              //left to complete later
+                              break;
+                      }
+
+                      Inlet_Values[nDim+2] = unit_r*cos(Theta) - unit_Theta*sin(Theta); //for ix
+                      Inlet_Values[nDim+3] = unit_r*sin(Theta) + unit_Theta*cos(Theta); //for iy
+                      Inlet_Values[nDim+4] = sqrt(1-pow(unit_r,2)- pow(unit_Theta,2));  //for iz
+
+                     solver[MESH_0][KIND_SOLVER]->SetInletAtVertex(Inlet_Values, iMarker, iVertex);                      
+                 
+                     for  (iVar=0; iVar < (maxCol_InletFile+nDim); iVar++)
+                        Inlet_InterpolatedData[iVertex*(maxCol_InletFile+nDim)+iVar]=Inlet_Values[iVar];
+                    
+                  }
+                  
+                  }
+                  if (Point_Match == false){
+                    unsigned long GlobalIndex = geometry[MESH_0]->node[iPoint]->GetGlobalIndex();
+                    cout<< "WARNING: Point in grid outside the inlet file radius range (extrapolation capability not yet added)" <<endl;
+                    cout<< "Point "<< GlobalIndex;
+                    cout<< std::scientific;
+                    cout << " at location: [" << Coord[0] << ", " << Coord[1] << ", Radius: "<<Interp_Radius;
+                    if (nDim ==3) error_msg << ", " << Coord[2];
+                    cout << "]" << endl;
+                    local_failure++;
+                    break;
+                }
+              }
+              if(Point_Match == true){  
+                
+                if(config->GetPrintInlet_InterpolatedData() == true)
+                   PrintNewInletData(Inlet_InterpolatedData,nVertex,Marker_Tag);
+                   
+                Point_Match = false;}
+            }
+          }
+        
+        }
+        if (local_failure > 0) break;
+      }
+      
+      maxCol_InletFile+=nDim;
+
+#ifdef HAVE_MPI
+    SU2_MPI::Allreduce(&local_failure, &global_failure, 1, MPI_UNSIGNED_SHORT,
+                       MPI_SUM, MPI_COMM_WORLD);
+#else
+    global_failure = local_failure;
+#endif
+
+    if (global_failure > 0) {
+      SU2_MPI::Error(string("Vertex radius outside inlet file radius range."), CURRENT_FUNCTION);
+    }
```

Comment:
I think you can just throw the error locally (without the reduction). And you do not need the #ifdef HAVE_MPI, you can write the code as if MPI is available as the functions are silently replaced for serial compilation.

Reply:
I don't really have any experience with coding for MPI. This part of the code I got from the existing "LoadInletProfile" subroutine and I kept it the way it was. For some reason, my code is failing the Reverse/Forward No/MPI regression tests (4 tests) although it runs fine with no MPI. Do you have an idea how can I fix this?

Reply:
From the compilation logs it seems like it is failing because of mixing passivedouble and su2double, try making everything su2double.

Reply:
You were right. That solved the problem.